### PR TITLE
Stats: Remove deprecated code

### PR DESCRIPTION
**Pull request by staskus**
After Stats Subscribers project there's plenty of code that is no longer used that should be removed:

- Stats Subscribers feature flag
- Classes that have Deprecated suffix
- Unused or changed Stats structure representing enums
- Unused data fetching in Stores
- Unused parameters in Views

## To test:

The only code that had to change a bit was related to Stats tabs enum and routing:

### Regression, persisting last selected tab:

1. Open Jetpack and login
2. Open Stats
3. Switch between tabs, leave, and return back
4. Confirm that the last selected tab is selected
5. Open Stats, Traffic, and switch to a different date period
6. Leave, and return back
7. Confirm that Traffic tab opens with previously selected date

### Regression, routing:

1. Open Jetpack and login
2. Add Day, and Week widgets
3. Confirm that Day widget opens Traffic with Day filter selected
4. Confirm that Week widget opens Traffic with Week filter selected

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/414f8e44-59a1-4f97-bf5b-1d45f8dfa89c

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/ac15d378-7dd2-4d9f-a42a-a63fcfbc8fe5

## Regression Notes
1. Potential unintended areas of impact

Making sure existing functionality works. Most of the areas are unaffected and I'm just removing competely unused code

3. What I did to test those areas of impact (or what existing automated tests I relied on)

4. What automated tests I added (or what prevented me from doing so)

Existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/Utility/Universal Links/Routes+Stats.swift [hunk 1]
```diff
@@ -95,19 +95,19 @@ extension StatsRoute: NavigationAction {
                 showStatsForDefaultBlog(from: values, with: coordinator)
             }
         case .daySite:
-            showStatsForBlog(from: values, tab: .days, using: coordinator)
+            showStatsForBlog(from: values, tab: .traffic, unit: .day, using: coordinator)
         case .weekSite:
-            showStatsForBlog(from: values, tab: .weeks, using: coordinator)
+            showStatsForBlog(from: values, tab: .traffic, unit: . week, using: coordinator)
```
Comment (guarani) [inline WordPress/Classes/Utility/Universal Links/Routes+Stats.swift]:
```suggestion
            showStatsForBlog(from: values, tab: .traffic, unit: .week, using: coordinator)
```

**Review by guarani — APPROVED** [review]
Thanks @staskus! 
I smoke tested the Traffic and Insight cards and their details screens and didn't find any issues.

File: WordPress/Classes/Utility/Universal Links/Routes+Stats.swift (hunk 1 above)
Comment (staskus) [reply WordPress/Classes/Utility/Universal Links/Routes+Stats.swift]: Thanks, strange that linter didn't catch it

**Pull request merged**